src/main/scala/org/ergoplatform/modifiers/history/Header.scala

```diff
@@ -128,83 +127,101 @@ object Header extends ApiCodecs {
   }
 }
 
-object HeaderSerializer extends Serializer[Header] {
-
-  def bytesWithoutInterlinksAndPow(h: Header): Array[Byte] =
-    Bytes.concat(
-      Array(h.version),
-      idToBytes(h.parentId),
-      h.ADProofsRoot,
-      h.transactionsRoot,
-      h.stateRoot,
-      Longs.toByteArray(h.timestamp),
-      h.extensionRoot,
-      RequiredDifficulty.toBytes(h.nBits),
-      Ints.toByteArray(h.height),
-      h.votes)
-
-  def bytesWithoutPow(h: Header): Array[Byte] = {
+object HeaderSerializer extends ScorexSerializer[Header] {
+
+  override def serialize(h: Header, w: Writer): Unit = {
+    serializeWithoutPow(h, w)
+    serializeSolution(h, w)
+  }
+
+  def serializeWithoutInterlinksAndPow(h: Header, w: Writer): Unit = {
+    w.put(h.version)
+    w.putBytes(idToBytes(h.parentId))
+    w.putBytes(h.ADProofsRoot)
+    w.putBytes(h.transactionsRoot)
+    w.putBytes(h.stateRoot)
+    w.putLong(h.timestamp)
+    w.putBytes(h.extensionRoot)
+    RequiredDifficulty.serialize(h.nBits, w)
+    w.putInt(h.height)
```

Comment:
VLQ might be useful here, height should not be too big (up to 1M in first 4 years)